The acquisition of the status of the emergency switch connected to the right motor is sometimes slow.

https://github.com/ezWheelSAS/swd_ros_controllers/issues/12
The status of the emergency switch connected to the left motor can now be acquired immediately.
However, acquiring the status of the emergency switch connected to the right motor takes time.
The trend is that it seems to be slower the first time it is pressed after startup.
After the second press, the status can be acquired relatively quickly.



```
fig@x86:/opt/ezw/usr/sbin$ ./remote.py swd_left
   1 : 	[8201] SRDO_SCT_timeout
   2 : 	[8201] SRDO_SCT_timeout
   3 : 	[8201] SRDO_SCT_timeout
   4 : 	[8201] SRDO_SCT_timeout
   5 : 	[8201] SRDO_SCT_timeout
   6 : 	[8201] SRDO_SCT_timeout
   7 : 	[8201] SRDO_SCT_timeout
   8 : 	[8201] SRDO_SCT_timeout
   9 : 	[8201] SRDO_SCT_timeout
  10 : 	[8201] SRDO_SCT_timeout
  11 : 	[8201] SRDO_SCT_timeout
  12 : 	[8201] SRDO_SCT_timeout
  13 : 	[8201] SRDO_SCT_timeout
  14 : 	[8201] SRDO_SCT_timeout
  15 : 	[8201] SRDO_SCT_timeout
  16 : 	[8201] SRDO_SCT_timeout
  17 : 	[8201] SRDO_SCT_timeout
  18 : 	[8201] SRDO_SCT_timeout
  19 : 	[8201] SRDO_SCT_timeout
  20 : 	[8201] SRDO_SCT_timeout
  21 : 	[8201] SRDO_SCT_timeout
  22 : 	[8201] SRDO_SCT_timeout
  23 : 	[8201] SRDO_SCT_timeout
  24 : 	[8201] SRDO_SCT_timeout
  25 : 	[8201] SRDO_SCT_timeout
  26 : 	[8201] SRDO_SCT_timeout
  27 : 	[8201] SRDO_SCT_timeout
  28 : 	[8201] SRDO_SCT_timeout
  29 : 	[8201] SRDO_SCT_timeout
  30 : 	[8201] SRDO_SCT_timeout
  31 : 	[8201] SRDO_SCT_timeout
  32 : 	[8201] SRDO_SCT_timeout
  33 : 	[8201] SRDO_SCT_timeout
  34 : 	[8201] SRDO_SCT_timeout
  35 : 	[8201] SRDO_SCT_timeout
  36 : 	[8201] SRDO_SCT_timeout
  37 : 	[8201] SRDO_SCT_timeout
  38 : 	[8201] SRDO_SCT_timeout
  39 : 	[8201] SRDO_SCT_timeout
  40 : 	[8201] SRDO_SCT_timeout
  41 : 	[8201] SRDO_SCT_timeout
  42 : 	[8201] SRDO_SCT_timeout
  43 : 	[8201] SRDO_SCT_timeout
  44 : 	[8201] SRDO_SCT_timeout
  45 : 	[8201] SRDO_SCT_timeout
  46 : 	[8201] SRDO_SCT_timeout
  47 : 	[8201] SRDO_SCT_timeout
  48 : 	[8201] SRDO_SCT_timeout
  49 : 	[8201] SRDO_SCT_timeout
  50 : 	[8201] SRDO_SCT_timeout
  51 : 	[8201] SRDO_SCT_timeout
  52 : 	[8201] SRDO_SCT_timeout
  53 : 	[8201] SRDO_SCT_timeout
  54 : 	[8201] SRDO_SCT_timeout
  55 : 	[8201] SRDO_SCT_timeout
  56 : 	[8201] SRDO_SCT_timeout
  57 : 	[8201] SRDO_SCT_timeout
  58 : 	[8201] SRDO_SCT_timeout
  59 : 	[8201] SRDO_SCT_timeout
  60 : 	[8201] SRDO_SCT_timeout
  61 : 	[8201] SRDO_SCT_timeout
  62 : 	[8201] SRDO_SCT_timeout
  63 : 	[8201] SRDO_SCT_timeout
  64 : 	[8201] SRDO_SCT_timeout
  65 : 	[8201] SRDO_SCT_timeout
  66 : 	[8201] SRDO_SCT_timeout
  67 : 	[8201] SRDO_SCT_timeout
  68 : 	[8201] SRDO_SCT_timeout
  69 : 	[8201] SRDO_SCT_timeout
  70 : 	[8201] SRDO_SCT_timeout
  71 : 	[8201] SRDO_SCT_timeout
  72 : 	[8201] SRDO_SCT_timeout
  73 : 	[8201] SRDO_SCT_timeout
  74 : 	[8201] SRDO_SCT_timeout
  75 : 	[8201] SRDO_SCT_timeout
  76 : 	[8201] SRDO_SCT_timeout
  77 : 	[8201] SRDO_SCT_timeout
  78 : 	[8201] SRDO_SCT_timeout
  79 : 	[8201] SRDO_SCT_timeout
  80 : 	[8201] SRDO_SCT_timeout
  81 : 	[8201] SRDO_SCT_timeout
  82 : 	[8201] SRDO_SCT_timeout
  83 : 	[8201] SRDO_SCT_timeout
  84 : 	[8201] SRDO_SCT_timeout
  85 : 	[8201] SRDO_SCT_timeout
  86 : 	[8201] SRDO_SCT_timeout
  87 : 	[8201] SRDO_SCT_timeout
  88 : 	[8201] SRDO_SCT_timeout
  89 : 	[8201] SRDO_SCT_timeout
  90 : 	[8201] SRDO_SCT_timeout
  91 : 	[8201] SRDO_SCT_timeout
  92 : 	[8201] SRDO_SCT_timeout
  93 : 	[8201] SRDO_SCT_timeout
  94 : 	[8201] SRDO_SCT_timeout
  95 : 	[8201] SRDO_SCT_timeout
  96 : 	[8201] SRDO_SCT_timeout
  97 : 	[8201] SRDO_SCT_timeout
  98 : 	[8201] SRDO_SCT_timeout
  99 : 	[8201] SRDO_SCT_timeout
 100 : 	[8201] SRDO_SCT_timeout
 101 : 	[8201] SRDO_SCT_timeout
 102 : 	[8201] SRDO_SCT_timeout
 103 : 	[8201] SRDO_SCT_timeout
 104 : 	[8201] SRDO_SCT_timeout
 105 : 	[8201] SRDO_SCT_timeout
 106 : 	[8201] SRDO_SCT_timeout
 107 : 	[8201] SRDO_SCT_timeout
 108 : 	[8201] SRDO_SCT_timeout
 109 : 	[8201] SRDO_SCT_timeout
 110 : 	[8201] SRDO_SCT_timeout
 111 : 	[8201] SRDO_SCT_timeout
 112 : 	[8201] SRDO_SCT_timeout
 113 : 	[8201] SRDO_SCT_timeout
 114 : 	[8201] SRDO_SCT_timeout
 115 : 	[8201] SRDO_SCT_timeout
 116 : 	[8201] SRDO_SCT_timeout
 117 : 	[8201] SRDO_SCT_timeout
 118 : 	[8201] SRDO_SCT_timeout
 119 : 	[8050] Generic_driver_error
 120 : 	[8201] SRDO_SCT_timeout
 121 : 	[800E] FAULT_ALIM_EXT

fig@x86:/opt/ezw/usr/sbin$ ./remote.py swd_right
   1 : 	[800E] FAULT_ALIM_EXT
   2 : 	[800E] FAULT_ALIM_EXT
   3 : 	[8201] SRDO_SCT_timeout
   4 : 	[800E] FAULT_ALIM_EXT
```

```
May 15 15:49:37 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597377.614382386]: NMT state machine is not OK.
May 15 15:49:37 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597377.614507412]: PDS state machine is not OK.
May 15 15:49:38 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597378.624806236]: NMT state machine is not OK.
May 15 15:49:38 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597378.625198025]: PDS state machine is not OK.
May 15 15:49:39 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597379.620269694]: NMT state machine is not OK.
May 15 15:49:39 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597379.620802020]: PDS state machine is not OK.
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: ;544;Controller;FAULT detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:54;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:54;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:54;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:54;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:54;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:55;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:56;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:57;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:58;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:58;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:58;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:58;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:58;;SMCDS;INFO;enterInOperationEnabledState;571;Controller;SWITCH_ON_DISABLED detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;591;Controller;READY_TO_SWITCH_ON detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;612;Controller;SWITCHED_ON detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 12:21:59;;SMCDS;INFO;enterInOperationEnabledState;636;Controller;OPERATION_ENABLED detected;
May 15 15:49:40 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:41 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;I[ERROR] [1652597381.758150738]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 15 15:49:41 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597381.758691456]: PDS state machine is not OK.
May 15 15:49:42 x86 sce-ros-bringup.sh[3546]: [ERROR] [1652597382.093811712]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 15 15:49:42 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597382.093901247]: PDS state machine is not OK.
May 15 15:49:43 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597383.654869236]: PDS state machine is not OK.
May 15 15:49:46 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597386.625760389]: NMT state machine is not OK.
May 15 15:49:46 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597386.625839625]: PDS state machine is not OK.
May 15 15:49:47 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597387.619671501]: NMT state machine is not OK.
May 15 15:49:47 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597387.619748496]: PDS state machine is not OK.
May 15 15:49:48 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597388.613182760]: NMT state machine is not OK.
May 15 15:49:48 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597388.613258152]: PDS state machine is not OK.
May 15 15:49:49 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597389.625035919]: NMT state machine is not OK.
May 15 15:49:49 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597389.625332597]: PDS state machine is not OK.
May 15 15:49:50 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597390.624128353]: NMT state machine is not OK.
May 15 15:49:50 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597390.624222617]: PDS state machine is not OK.
May 15 15:49:51 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597391.620330520]: NMT state machine is not OK.
May 15 15:49:51 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597391.620407638]: PDS state machine is not OK.
May 15 15:49:52 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597392.621758923]: NMT state machine is not OK.
May 15 15:49:52 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597392.622063410]: PDS state machine is not OK.
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: NFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;INFO;enterInOperationEnabledState;571;Controller;SWITCH_ON_DISABLED detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:40;;SMCDS;INFO;enterInOperationEnabledState;591;Controller;READY_TO_SWITCH_ON detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;612;Controller;SWITCHED_ON detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;636;Controller;OPERATION_ENABLED detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:41;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:42;;SMCDS;INFO;enterInOperationEnabledState;571;Controller;SWITCH_ON_DISABLED detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;591;Controller;READY_TO_SWITCH_ON detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;612;Controller;SWITCHED_ON detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:43;;SMCDS;INFO;enterInOperationEnabledState;636;Controller;OPERATION_ENABLED detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;571;Controller;SWITCH_ON_DISABLED detected;
May 15 15:49:53 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 15 15:49:54 x86 sce-ros-bringup.sh[3546]: 2022-5-15 15:49:53;;SMCDS;INFO;enterInOperationEnabledState;591;Controller;READY[ERROR] [1652597394.766509639]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 15 15:49:54 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597394.766581057]: PDS state machine is not OK.
May 15 15:49:55 x86 sce-ros-bringup.sh[3546]: [ERROR] [1652597395.097346374]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 15 15:49:55 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597395.097485203]: PDS state machine is not OK.
May 15 15:49:55 x86 sce-ros-bringup.sh[3546]: [ERROR] [1652597395.930099630]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 15 15:49:55 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597395.930173480]: PDS state machine is not OK.
May 15 15:49:57 x86 sce-ros-bringup.sh[3546]: [ WARN] [1652597397.658740483]: PDS state machine is not OK
```

```
May 23 10:21:22 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268882.519135503]: PDS state machine is not OK.
May 23 10:21:23 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268883.519838798]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:23 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268883.519908396]: PDS state machine is not OK.
May 23 10:21:24 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268884.516278703]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:24 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268884.516348180]: PDS state machine is not OK.
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: roller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:20;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:20;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:21;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:22;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:23;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:24;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:25;;SMCDS;ERROR;enterInOperationEnabledState[ERROR] [1653268885.516044351]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabled>
May 23 10:21:25 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268885.516103812]: PDS state machine is not OK.
May 23 10:21:26 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268886.517162188]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:26 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268886.517234833]: PDS state machine is not OK.
May 23 10:21:27 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268887.512842722]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:27 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268887.512913118]: PDS state machine is not OK.
May 23 10:21:28 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268888.521685261]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:28 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268888.521757841]: PDS state machine is not OK.
May 23 10:21:29 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268889.509180939]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:29 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268889.509486455]: PDS state machine is not OK.
May 23 10:21:30 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268890.520530617]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:30 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268890.520801428]: PDS state machine is not OK.
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: ;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:26;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:27;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:28;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:29;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;535;Controller;Check FAULT;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:30;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_ON;
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInO[ERROR] [1653268891.513971647]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error >
May 23 10:21:31 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268891.514041706]: PDS state machine is not OK.
May 23 10:21:32 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268892.516161216]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:32 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268892.516228486]: PDS state machine is not OK.
May 23 10:21:33 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268893.509522627]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:33 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268893.509595340]: PDS state machine is not OK.
May 23 10:21:34 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268894.518527444]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:34 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268894.518596433]: PDS state machine is not OK.
May 23 10:21:35 x86 sce-ros-bringup.sh[3595]: [ERROR] [1653268895.519926182]: Failed to set PDS state for right motor, EZW_ERR: SMCService : Controller::enterInOperationEnabledState() return error code : 6
May 23 10:21:35 x86 sce-ros-bringup.sh[3595]: [ WARN] [1653268895.520023973]: PDS state machine is not OK.
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: perationEnabledState;535;Controller;Check FAULT;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;544;Controller;FAULT detected;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;562;Controller;Check SWITCH_ON_DISABLED;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;582;Controller;Check READY_TO_SWITCH_ON;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;603;Controller;Check SWITCHED_ON;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;INFO;enterInOperationEnabledState;623;Controller;Check OPERATION_ENABLED;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:31;;SMCDS;ERROR;enterInOperationEnabledState;632;Controller;Not in OPERATION_ENABLED;
May 23 10:21:36 x86 sce-ros-bringup.sh[3595]: 2022-5-23 10:21:32;;SMCDS;INFO;enterInOperationEnabledState;515;Controller;Check NOT_READY_TO_SWITCH_O
```

--- a/include/diff_drive_controller/DiffDriveController.hpp
+++ b/include/diff_drive_controller/DiffDriveController.hpp
@@ -11,6 +11,8 @@
 #include "ezw-smc-core/Config.hpp"
 #include "ezw-smc-core/Controller.hpp"
 
+#include "ezw-canopen-service/DBusClient.hpp"
+
 #include <swd_ros_controllers/SafetyFunctions.h>
 
 #include <geometry_msgs/Point.h>
@@ -66,8 +68,9 @@ namespace ezw
             std::string m_odom_frame, m_base_frame, m_left_config_file, m_right_config_file;
             bool        m_have_backward_sls, m_publish_odom, m_publish_tf, m_publish_safety, m_nmt_ok, m_pds_ok;
 
-            ros::Timer               m_timer_odom, m_timer_watchdog, m_timer_pds, m_timer_safety;
-            ezw::smccore::Controller m_left_controller, m_right_controller;
+            ros::Timer                      m_timer_odom, m_timer_watchdog, m_timer_pds, m_timer_safety;
+            ezw::smccore::Controller        m_left_controller, m_right_controller;
+            ezw::canopenservice::DBusClient m_cos_client;
 
             std::mutex                           m_safety_msg_mtx;
             swd_ros_controllers::SafetyFunctions m_safety_msg;

--- a/src/diff_drive_controller/DiffDriveController.cpp
+++ b/src/diff_drive_controller/DiffDriveController.cpp
@@ -20,6 +20,8 @@
 #include <limits>
 #include <tf2/LinearMath/Quaternion.h>
 
+#include <unistd.h>
+
 using namespace std::chrono_literals;
 
 #define USE_SAFETY_CONTROL_WORD 0
@@ -266,8 +268,18 @@ namespace ezw
                      "Setting maximum motor safety limited speed to %d rpm",
                      max_sls_wheel_speed_rpm, m_motor_sls_rpm);
 
+            // Init DBus COS client
+            err = m_cos_client.init();
+            if (ERROR_NONE != err) {
+                ROS_ERROR("Failed initializing COS client : "
+                          "ezw::canopenservice::DBusClient::init() return error code : %d",
+                          (int)err);
+                throw std::runtime_error("Failed initializing COS client");
+            }
+
+            // Create timers
             m_timer_watchdog = m_nh->createTimer(ros::Duration(m_watchdog_receive_ms / 1000.0), boost::bind(&DiffDriveController::cbWatchdog, this));
-            m_timer_pds      = m_nh->createTimer(ros::Duration(1.0), boost::bind(&DiffDriveController::cbTimerStateMachine, this));
+            m_timer_pds      = m_nh->createTimer(ros::Duration(0.5), boost::bind(&DiffDriveController::cbTimerStateMachine, this));
 
             if (m_publish_odom || m_publish_tf) {
                 m_timer_odom = m_nh->createTimer(ros::Duration(1.0 / m_pub_freq_hz), boost::bind(&DiffDriveController::cbTimerOdom, this));
@@ -305,24 +317,22 @@ namespace ezw
                           (int)err_r);
             }
 
-            if (smccore::Controller::NMTState::OPER != nmt_state_l) {
-                err_l = m_left_controller.setNMTState(smccore::Controller::NMTCommand::OPER);
-            }
-
-            if (smccore::Controller::NMTState::OPER != nmt_state_r) {
-                err_r = m_right_controller.setNMTState(smccore::Controller::NMTCommand::OPER);
-            }
-
-            if (ERROR_NONE != err_l && smccore::Controller::NMTState::OPER != nmt_state_l) {
-                ROS_ERROR("Failed to set NMT state for left motor, EZW_ERR: SMCService : "
-                          "Controller::setNMTState() return error code : %d",
-                          (int)err_l);
-            }
-
-            if (ERROR_NONE != err_r && smccore::Controller::NMTState::OPER != nmt_state_r) {
-                ROS_ERROR("Failed to set NMT state for right motor, EZW_ERR: SMCService : "
-                          "Controller::setNMTState() return error code : %d",
-                          (int)err_r);
+            if (smccore::Controller::NMTState::OPER != nmt_state_l || smccore::Controller::NMTState::OPER != nmt_state_r) {
+                // NodeId = 0 => broadcast to all canopen nodes
+                int err = m_cos_client.setNmtState(0, ezw_nmt_command_t::EZW_CO_NMT_COMMAND_PREOP);
+                if (ERROR_NONE != err) {
+                    ROS_ERROR("Failed to broadcast NMT state EZW_CO_NMT_COMMAND_RESET_COMM"
+                              "ezw::canopenservice::DBusClient::setNMTState() return error code : %d",
+                              (int)err);
+                } else {
+                    usleep((10) * 1000);
+                    err = m_cos_client.setNmtState(0, ezw_nmt_command_t::EZW_CO_NMT_COMMAND_OPER);
+                    if (ERROR_NONE != err) {
+                        ROS_ERROR("Failed to broadcast NMT state OPER"
+                                  "ezw::canopenservice::DBusClient::setNMTState() return error code : %d",
+                                  (int)err);
+                    }
+                }
             }
 
             m_nmt_ok = (smccore::Controller::NMTState::OPER == nmt_state_l) && (smccore::Controller::NMTState::OPER == nmt_state_r);

